lit_gpt/prompts.py

```diff
@@ -61,11 +72,270 @@ def apply(self, prompt: str, **kwargs: str) -> str:
+    def apply(self, prompt: str, **kwargs: str) -> str:
+        return f"### Instruction:\n{prompt}\n\n### Response:\n"
+
+
+class StableCode(PromptStyle):
+    def apply(self, prompt: str, **kwargs: str) -> str:
+        return f"###Instruction\n{prompt}###Response\n"
+
+
+class CodeLlama(PromptStyle):
+    def apply(self, prompt: str, **kwargs: str) -> str:
+        # for CodeLLama, we don't set a default system prompt, but it is supported:
+        # https://huggingface.co/blog/codellama#conversational-instructions
+        # Mistral does not: https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.1#instruction-format
+        b_inst, e_inst = "<s>[INST]", "[/INST]"
+        return f"{b_inst} {prompt} {e_inst}"
+
+
+class Phi1(PromptStyle):
+    def apply(self, prompt: str, **kwargs: str) -> str:
+        return f"{prompt}\n\nAnswer:"
+
+    def stop_tokens(self, tokenizer: "Tokenizer") -> Tuple[List[int], ...]:
+        return (
+            [tokenizer.eos_id],
+            [tokenizer.token_to_id("Answer"), tokenizer.token_to_id(":")],
+            [198, tokenizer.token_to_id("Answer"), tokenizer.token_to_id(":")],
+            # the model rarely emits the eos token and instead outputs newlines, but we cannot use them
+            # to stop or else things like code generation wouldn't work
+            # [198, 198],  # '\n', '\n'
+        )
+
+
+class Phi2(PromptStyle):
+    def apply(self, prompt: str, **kwargs: str) -> str:
+        return f"Instruct:{prompt}\nOutput:"
+
+
+class TinyLlama(PromptStyle):
+    def apply(self, prompt: str, **kwargs: str) -> str:
+        return (
+            "<|system|>\n"
+            "You are a friendly chatbot who always gives helpful, detailed, and polite answers.</s>\n"
+            "<|user|>\n"
+            f"{prompt}</s>\n"
+            "<|assistant|>\n"
+        )
+
+
+class Gemma(PromptStyle):
+    def apply(self, prompt: str, **kwargs: str) -> str:
+        return f"<start_of_turn>user\n{prompt}<end_of_turn>\n<start_of_turn>model\n"
+
+
 # Maps prompt style names to PromptStyle classes
 prompt_styles: Dict[str, Type[PromptStyle]] = {
     # Dataset-specific prompt styles
     "alpaca": Alpaca,
     "flan": FLAN,
     "longform": Longform,
-    # TODO: Model-specific prompt styles
+    # Model-specific prompt styles
+    "stablelm-alpha": StableLMAlpha,
+    "stablelm-zephyr": StableLMZephyr,
+    "togethercomputer-chat": TogetherComputerChat,
+    "togethercomputer-instruct": TogetherComputerInstruct,
+    "falcon": Falcon,
+    "vicuna": Vicuna,
+    "llama2-function-calling": Llama2FunctionCalling,
+    "llama2": Llama2,
+    "freewilly2": FreeWilly2,
+    "platypus": Platypus,
+    "nous-research": NousResearch,
+    "stablecode": StableCode,
+    "codellama": CodeLlama,
+    "phi-1": Phi1,
+    "phi-2": Phi2,
+    "tinyllama": TinyLlama,
+    "gemma": Gemma,
+}
+
+# Maps HF model names to prompt style names
+model_name_to_prompt_style = {
+    "stablelm-tuned-alpha-3b": "stablelm-alpha",
```

Comment:
What's the benefit of the step in between, couldn't this use `StableLMAlpha` as the value?